modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js

```diff
@@ -67,6 +67,20 @@ export default function defaultFetch(resource, init = {}) {
 				resource = {...resource, url: resourceLocation};
 			}
 		}
+
+		const doAsGroupId = Liferay.ThemeDisplay.getScopeGroupId();
+
+		if (doAsGroupId) {
+			resourceURL.searchParams.set('doAsGroupId', doAsGroupId);
+			resourceLocation = resourceURL.toString();
+
+			if (typeof resource === 'string') {
+				resource = resourceLocation;
+			}
+			else {
+				resource = {...resource, url: resourceLocation};
+			}
```

Comment:
IS there any reason why we need to repeat the block?:

```
			resourceLocation = resourceURL.toString();

			if (typeof resource === 'string') {
				resource = resourceLocation;
			}
			else {
				resource = {...resource, url: resourceLocation};
			}
```

It's 3 times there. Looks like its primary goal is to  update `resourceLocation` and `resourceURL` but the part concerning `resource` could be moved to be run right before the actual `fetch` call right? cc/ @bryceosterhaus  perhaps I'm not seeing something here, let me know

Reply:
Yes, actually is what we do in liferay-frontend#3395. I'll just rebase this on top of that one.

Reply:
great!